sdk/core/azure-core/src/http/curl/curl_connection_private.hpp

```diff
@@ -42,6 +42,33 @@ namespace Azure { namespace Core {
     {
       using type = _internal::BasicUniqueHandle<CURL, curl_easy_cleanup>;
     };
+
+    /**
+     *
+     * @brief Unique handle wrapper for CURLSH handles.
+     *
+     * @note Internally, CURL and CURLSH are declared as the same void type,
+     * so to avoid collisions we need this wrapper.
+     */
+    struct CURLSHWrapper
+    {
+      CURLSH* share_handle;
+
+      CURLSHWrapper() : share_handle{curl_share_init()} {};
+    };
+
+    /**
+     * @brief   Unique handle for CURLSHWrapper handles
+     */
+    template <> struct UniqueHandleHelper<CURLSHWrapper>
+    {
+      static void FreeCurlShare(CURLSHWrapper* curl_share)
+      {
+        curl_share_cleanup(curl_share->share_handle);
+        free(curl_share);
+      }
+      using type = _internal::BasicUniqueHandle<CURLSHWrapper, FreeCurlShare>;
```

Comment:
I think you either use UniqueHandleHelper with CURLSH + FreeCurlShare() which invokes curl_share_cleanup(), or you make your own wrapper, CURLSHWrapper, which allocates it in the constructor, and calls curl_share_cleanup() in destructor, and then you don't need the UniqueHandleHelper - a regualr unique_ptr will do.

Reply:
And then you can do `using UniqueCURLSHHandle = UniqueHandle<CURLSH, FreeCurlShare>;`, and then you declare `UniqueCURLSHHandle m_sslShareHandle;`, so there is no confusion between `void*` for the compiler, because in the typedef you provided the second argument for the UniqueHandle.